src/main/java/frc/robot/scoreStates/ScoreStateConstants.java

```diff
@@ -22,20 +20,53 @@ public class ScoreStateConstants {
-    public static final List<Pose2d> OPTIMAL_POINTS_TRAP_RED =
-            OPTIMAL_POINTS_TRAP_BLUE.stream()
+    public static final List<Pose2d> OPTIMAL_POINTS_CLIMB_RED =
+            OPTIMAL_POINTS_CLIMB_BLUE.stream()
                     .map(GeometryUtil::flipFieldPose)
                     .collect(Collectors.toList());
     public static final Translation2d AMP_POSE_BLUE = new Translation2d(1.85, 7.49);
-    public static final Translation2d AMP_POSE_RED = GeometryUtil.flipFieldPosition(AMP_POSE_BLUE);
+    public static final Translation2d AMP_POSE_RED =
+            new Translation2d(0.0, 0.0); // TODO: add real value
     public static final Translation2d SPEAKER_POSE_BLUE = new Translation2d(0, 5.547_944_2);
     public static final Translation2d SPEAKER_POSE_RED =
             GeometryUtil.flipFieldPosition(SPEAKER_POSE_BLUE);
     public static final Rotation2d AMP_ROTATION_NORMAL = new Rotation2d(Math.toRadians(90));
     public static final Rotation2d AMP_ROTATION_REVERSE = new Rotation2d(Math.toRadians(-90));
-    public static final Measure<Distance> MAX_SHOOTING_DISTANCE = Units.Meters.of(3);
-    public static final Measure<Distance> MIN_DISTANCE_TO_TURN_GRIPPER = Units.Meters.of(3);
-    public static final Measure<Angle> TURN_TOLERANCE = Units.Rotations.of(0.01); // Rotations
+    public static final MutableMeasure<Distance> MIN_DISTANCE_TO_TURN_GRIPPER =
+            Units.Meters.of(3).mutableCopy();
+    public static final MutableMeasure<Angle> TURN_TOLERANCE = Units.Degrees.of(5).mutableCopy();
+
+    public static final double SPEAKER_TARGET_HEIGHT = 0.0; // TODO: check real value
+
+    public static final InterpolatingDoubleMap BLUE_BOUNDS_MAP = new InterpolatingDoubleMap(6);
+    public static final InterpolatingDoubleMap RED_BOUNDS_MAP = new InterpolatingDoubleMap(6);
+
+    static {
+        BLUE_BOUNDS_MAP.putAll(
+                new HashMap<>() {
+                    {
+                        for (int i = 0; i < OPTIMAL_POINTS_SHOOT_BLUE.size(); i++) {
+                            put(
+                                    new InterpolatingDouble(
+                                            OPTIMAL_POINTS_SHOOT_BLUE.get(i).getY()),
+                                    new InterpolatingDouble(
+                                            OPTIMAL_POINTS_SHOOT_BLUE.get(i).getX()));
+                        }
+                    }
+                });
+
+        RED_BOUNDS_MAP.putAll(
+                new HashMap<>() {
+                    {
+                        for (int i = 0; i < OPTIMAL_POINTS_SHOOT_RED.size(); i++) {
+                            put(
+                                    new InterpolatingDouble(OPTIMAL_POINTS_SHOOT_RED.get(i).getY()),
+                                    new InterpolatingDouble(
+                                            OPTIMAL_POINTS_SHOOT_RED.get(i).getX()));
+                        }
+                    }
+                });
```

Comment:
These two are identical and should be exported to a function (it's fine for this function to be in the file).

Reply:
b772c71

Reply:
This can be more concise using the stream API:
```java
private static InterpolatingDoubleMap createBoundsMap(List<Translation2d> optimalPoints) {
    InterpolatingDoubleMap map = new InterpolatingDoubleMap();
    optimalPoints.forEach(point ->
            map.put(new InterpolatingDouble(point.getY()), new InterpolatingDouble(point.getX())));
    return map;
}
```

Reply:
502142a